### ISSUE TITLE
Usage of `title` via prop drilling not detected

### ISSUE THREAD

**Issue by theerebuss** [github/eslint-plugin-github#467]
Recently found a couple of cases of `title` attribute usage (soon to be addressed in https://github.com/github/issues/issues/7066 🙏) that are not detected by the rule.

This seems to be the case when prop drilling `title` into a component, unsure if because this specific scenario is overriding the type prop or something else.

[Root usage of the attribute](https://github.com/github/github/blob/3e4881e7aab7d0473ed9c1b89a3712a9e1d65f3b/ui/packages/issue-viewer/components/sections/RelationsSection.tsx#L84)
[Props object with the `title` prop being spread into a button](https://github.com/github/github/blob/3e4881e7aab7d0473ed9c1b89a3712a9e1d65f3b/ui/packages/issue-viewer/components/sections/SectionHeader.tsx#L48)
[Prop type overriding `title`](https://github.com/github/github/blob/3e4881e7aab7d0473ed9c1b89a3712a9e1d65f3b/ui/packages/issue-viewer/components/sections/SectionHeader.tsx#L12)

**Comment by kendallgassner** [comment]
This could be challenging because the `jst-ast-util` function we use to grab the `title` attribute only checks the direct keys.

**Comment by andrefcdias** [comment]
Agreed, maybe another approach could be to throw warning on usage of `title` directly in Primer components? We'd still have problems detecting prop drilling for regular JSX elements, but it could give another layer of security.

**Comment by kendallgassner** [comment]
cc. @TylerJDev I wonder if we can just omit the title props from this [Primer component](https://github.com/github/github/blob/3e4881e7aab7d0473ed9c1b89a3712a9e1d65f3b/ui/packages/issue-viewer/components/sections/SectionHeader.tsx#L8C1-L16C11)?

**Comment by TylerJDev** [comment]
@kendallgassner, In this example yeah! I think that the title prop should be removed from `IconProps`. I'm not entirely sure why it's present here, as it seems to be adding the native `title` attribute to the `Button` which doesn't seem correct.. This is what I'm thinking it looks like when rendered:

```.js
<button title="Passed text" ...>
    <span>Edit Passed Text</span>
</button>
```

We'd definitely want to remove it in this instance. I don't think this is tied to Primer though. I'm wondering, is this a shared component?

**Comment by andrefcdias** [comment]
This particular instance was already fixed in https://github.com/github/github/pull/281501 ✅

`SectionHeader` is just a shared component used for the metadata sections.

**Comment by andrefcdias** [comment]
That and the awareness raised within my team should be more than enough because we own the component in question and the shared component that uses it. My original concern was more about this scenario being a possibility, which can go under the radar as it is not direct `title` usage.

**Comment by kendallgassner** [comment]
@TylerJDev thoughts here? 

We have some Primer components that use the prop name `title`; but the `title` prop in these components does not get used as a semantic title attribute in the rendered HTML element. It be nice to avoid using this prop name but I don't think we are likely to change these props name since it might require a `major` version bump?

**Comment by andrefcdias** [comment]
> We have some Primer components that use the prop name title; but the title prop in these components does not get used as a semantic title attribute in the rendered HTML element.

FYI, I've also seen this happen in some of the shared components

**Comment by kendallgassner** [comment]
`eslint-plugin-github` could only test against semantic HTML components because we didn't want to flag react components that might use the `title` attribute differently. 

If we add the lint rule in `eslint-plugin-primer-react` we could expand are testing to not only flag semantic components but to also check against the primer specific components  -- since we have more control of these components. 

Thinking 🤔 though we would need to account for the `as` prop.

**Comment by TylerJDev** [comment]
If we did test Primer specific components, would we want to ignore `as` usage and still throw a violation regardless of what value it might have? I'm wondering, are there cases where `title` might be appropriate outside of `iframe`?

**Comment by kendallgassner** [comment]
And I can't think of a case where we would want a Primer component with `as=iframe` _except_ maybe `Box`.

**Comment by khiga8** [comment]
> Definitely agree, I think it'd be beneficial for us to steer away from using HTML attributes as prop names when there's no direct connection. This would involve a breaking change for most components that currently do this, but I'd like to get the idea out there.

Revisiting this discussion... @tylerjdev I 100% agree and would fully support this! 

Is there a tracking issue or discussion I could reference? :)

**Comment by byt3n33dl3** [comment]
Agreed, maybe another approach 🧛🏻
